LibertyDSNP/example-client
# pick handle at login

Purpose
---------------
It's possible for the same wallet address to be authorized as a delegate to multiple DSNP ids, so logging in to a wallet is not necessarily sufficient to determine which account to display in the example client.


Solution
---------------
This PR adds a modal to disambiguate the DSNP id and handle when there are multiple registrations for a wallet address. We were already retrieving all the registrations. This PR is just adds UI to pick between them rather than automatically picking the first one.

Change summary
---------------
* Add a second registration to the populate script for validation (and refactor a bit to permit accounts without posts).
* Modify the Login component to hold off on login and show the new RegistrationModal when multiple registrations are received.
* Implement registration modal to select registration.
* Style registration modal.


Steps to Verify
----------------
1. Set up app (hardhat, deploy, static server) and run populate.
2. Start app and log in with metamask using the `0x8626f6940e2eb28930efb4cef49b2d1f2c9c1199` account.
3. Expect a modal to appear showing two registrations.
4. Select one of the registrations and click "Select Handle".
5. Expect the app to be logged in and the account to match the selected registration.
6. Log out and log in again with the other registration.
7. Expect the account to be changed.


Additional details / screenshot
----------------
- <img width="405" alt="Screen Shot 2021-08-19 at 3 08 59 PM" src="https://user-images.githubusercontent.com/169613/130129713-223e858d-29c1-4f4b-a6a3-9bd9baaf7553.png">
